### DISCUSSION TITLE
Enum doesn't work/validate, when min_length in query parameters is defined

### DISCUSSION THREAD

**Discussion by eXTure** [fastapi/fastapi#8508] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.params import Query
import enum
from typing import List

app = FastAPI()


class RoomNumberEnum(str, enum.Enum):
    value_1 = "1"
    value_2 = "2"
    value_3 = "3"
    value_4 = "4"


@app.get("/")
def root(
        room_no_category: List[RoomNumberEnum] = Query(default=..., min_length=1)
):
    return {"message": f"Room number category: {room_no_category[0]}"}
```

### Description

If min_length is declared in parameters, RoomNumberEnum doesn't validate. If I delete min_length, enum validation works.

### Operating System

Windows

### FastAPI Version

0.78.0

### Python Version

3.7.7

**Comment by aboubacs** [comment]
What is your query and your error message ?

**Comment by eXTure** [comment]
@aboubacs There's no error message, the input just passes through without any errors, even though it's incorrect and not defined in enum class. It should be caught by enum, in case a value is not defined.

The min_length parameter is messing up the validation.

When looking at the api docks, the values are not visible, when min_length is defined.
`room_no_category: List[RoomNumberEnum] = Query(default=..., min_length=1)`

![image](https://user-images.githubusercontent.com/24888145/171229548-ac0b0e6f-19e0-4872-843d-8e764338eecb.png)


If I remove min_length from Query parameters, then it works as expected.
`room_no_category: List[RoomNumberEnum] = Query(default=...)`

![image](https://user-images.githubusercontent.com/24888145/171229792-44f3cc4d-c1ef-4f0c-bd23-c8156b263584.png)

**Comment by odiseo0** [comment]
The `min_length` parameter is because is a `list` and is validating the `list` instead of its content, you should make a custom validator instead of using `Query`.

Do you want a `list` of values and use the RoomNumberEnum to validate the content of the `list` or do you want only a number?

**Comment by eXTure** [comment]
@odiseo0 I need a list of values that are validated by the RoomNumberEnum. It works without min_length, so it's not a big issue, I thought it's a bug and wanted to report.

**Comment by YuriiMotov** [comment, marked as answer]
Works fine in fastapi 0.115.12
